kyaml/openapi/openapi.go

```diff
@@ -536,7 +537,13 @@ func parseBuiltinSchema(version string) {
 // parse parses and indexes a single json schema
 func parse(b []byte) error {
 	var swagger spec.Swagger
-
+	s := string(b)
+	if len(s) > 0 && s[0] != '{' {
+		j, err := yaml.YAMLToJSON(b)
+		if err == nil {
```

Comment:
Should we really attempt to unmarshal the original bytes if this failed? The doc for `yaml.YAMLToJSON` claims that it is a no-op for JSON, so if we get an error here, the problem shouldn't be that we attempted to convert JSON when we shouldn't have.

Reply:
Thanks for the catch, I updated it to return the error if there is one.